Sequences/MRML/vtkMRMLSequenceNode.cxx

```diff
@@ -256,7 +259,19 @@ void vtkMRMLSequenceNode::SetDataNodeAtValue(vtkMRMLNode* node, const char* inde
     return;
   }
 
+  std::string newNodeName = this->SequenceScene->GetUniqueNameByString( node->GetName() );
   vtkMRMLNode* newNode=this->SequenceScene->CopyNode(node);
+  newNode->SetName( newNodeName.c_str() ); // Make sure all the node names in the sequence's scene are unique for saving purposes
+  
+  // Hack to enforce deep copying of volumes (otherwise, a separate image data is not stored for each frame)
+  // TODO: Is there a better way to architecture this?
+  vtkMRMLVolumeNode* volNode = vtkMRMLVolumeNode::SafeDownCast(newNode);
+  if (volNode!=NULL)
+  {
+    vtkImageData* imageDataCopy = vtkImageData::New();
```

Comment:
Is something like this https://github.com/mholden8/Sequences/commit/eb7de1c3d17aabb9ea8b67398544c806315ae759 what you had in mind?